src/common/snippets/src/generator.cpp

```diff
@@ -44,8 +44,14 @@ void Generator::generate(lowered::LinearIR& linear_ir, LoweringResult& result, c
     }
     OV_ITT_TASK_NEXT(GENERATE, "::GetSnippet")
 
-    // Note: some emitters use precompiled kernels. They need to be saved, so the kernels are accessible at runtime.
-    if (linear_ir.get_config().m_save_expressions) {
+    if (target->is_err_detector_enabled()) {
+        // Need save emitters to print usefull info in runtime if segfault happens.
+        for (const auto& expr : linear_ir) {
+            const auto& emitter = expr->get_emitter();
+                result.m_saved_emitters.emplace_back(emitter);
+        }
+    } else if (linear_ir.get_config().m_save_expressions) {
+        // Note: some emitters use precompiled kernels. They need to be saved, so the kernels are accessible at runtime.
```

Comment:
I think it's better to handle the case with segfault detector enabled inside `uses_precompiled_kernel` and not introduce a new API for target machine.

I also discussed it offline with @IvanNovoselov: we agreed that it's better to remove `m_save_expressions` member from `linear_ir` and always check emmiters of `linear_ir`.
```
        for (const auto& expr : linear_ir) {
            const auto& emitter = expr->get_emitter();
            if (uses_precompiled_kernel(emitter))
                result.m_saved_emitters.emplace_back(emitter);
        }
```

Reply:
ok, only m_saved_emitters is preserved and m_save_expressions also deleted, thanks!